src/Symfony/Component/Workflow/Registry.php

```diff
@@ -23,15 +24,19 @@ class Registry
     private $workflows = array();
 
     /**
-     * @param Workflow                 $workflow
+     * @param Workflow                         $workflow
      * @param SupportStrategyInterface $supportStrategy
+     *
+     * @deprecated since Symfony 4.1. Use addWorkflow() instead.
```

Comment:
The path is technically correct, but is suggesting another method really the right thing to do? What if in the future `addWorkflow` gets flawed.. we suggest `add...Workflow...`?

Point is we probably impact more people calling `add()` today then people actually overriding `add()`, at least that's my guess/concern here. And `add()` is really the ideal name also IMHO.

Cant we simply remove the type declaration; add new API as type-hint/PHPDoc; and trigger a deprecation if the method is overridden with the _wrong_ type declaration? Isnt that more friendly?